editor/src/components/editor/actions/actions.tsx

```diff
@@ -1961,6 +1961,9 @@ export const UPDATE_FNS = {
     derived: DerivedState,
   ): EditorModel => {
     // same as UPDATE_EDITOR_MODE, but clears the drag state
+    if (action.unlessMode === editor.mode.type) {
```

Comment:
I think instead of the `unlessMode` flag, `clearDragState` should be turned into a full-fledged action, and whenever `action.unlessMode` would be the same as `editor.mode.type`, the new `clearDragState` action should be called.

The reason for this would be that with `unlessMode`, the `SWITCH_EDITOR_MODE` action could be misused in a way that wouldn't actually switch the editor mode. While this is not a huge deal, solving this via composition, rather than adding a flag to the existing action would lead to a cleaner implementation.

Reply:
turns out this isn't feasible in this PR (see https://github.com/concrete-utopia/utopia/pull/3031/commits/dbf885e89856e0cd7270d28c9ed8ea2a9e4c4160)